### ISSUE TITLE
Feature request: Add support for a more standard way of command line argument passing which enables 'Tab Completion'

### ISSUE THREAD

**Issue by lzhoucs** [theintern/intern#646]
When typing in the command line to specify the config file, it would be nice to have 'Tab Completion':
```
node_modules/.bin/intern-client config=tests/intern
```
Currently the completion is restricted by the fact the config path is not started as an argument. So I wonder if we can add support to the following style which seems pretty standard and will enable 'Tab Completion' naturally:
```
node_modules/.bin/intern-client --config tests/intern
```

**Comment by jason0x43** [comment]
In either case you would need a shell-specific completion script to get completion working, unless the CLI also emits a help message detailing all the arguments so a completion system can learn them.

**Comment by lzhoucs** [comment]
I just want the file path part of it to be 'auto-expandable', currently it is blocked by the fact it starts as part of an existing argument `config=` instead of a fresh one. So as a workaround, current I have to  type the file path first which is 'auto-expandable':
```
node_modules/.bin/intern-client tests/intern 
```
then manually go back and and `config=`:
```
node_modules/.bin/intern-client config=tests/intern
```

**Comment by jason0x43** [comment]
Ah, yes. That may end up being confusing, though, since the config parameter isn't a file path, it's a module ID. While those may look similar, they are not the same (a valid module ID doesn't have to mirror a real path).

**Comment by sholladay** [comment]
@jason0x43 is that actually true still? I know there was some back and forth on this.

In particular, there is [this commit](https://github.com/theintern/intern/commit/c81a1d19b604a034eae8b9f5378151c8f0518553) and related discussion in https://github.com/theintern/intern/issues/409#issuecomment-111536458.

> `config` is a module ID. ... I’m investigating just resolving the full path explicitly when retrieving the config since the module ID-ness of the config argument seems to baffle most.

**Comment by jason0x43** [comment]
@sholladay You are correct; I'm living in the past...

**Comment by jason0x43** [comment]
This now exists when using https://github.com/theintern/intern-cli
